prisma/schema.prisma

```diff
@@ -988,12 +1013,22 @@ model Image {
   engagements     ImageEngagement[]
   collectionItems CollectionItem[]
   collections     Collection[]
+  connections     ImageConnection[]
 
   @@index([featuredAt])
   @@index([postId], type: Hash)
   @@index([userId, postId])
 }
 
+model ImageConnection {
+  imageId    Int
+  image      Image  @relation(fields: [imageId], references: [id], onDelete: Cascade)
+  entityId   Int
+  entityType String
+
+  @@id([imageId, entityId, entityType])
```

Comment:
If we ever are to query images connected to entities than we would want to have a reverse lookup index on [entityType, entityId]